### PR TITLE
feat: refactor init of dtensor policy v2

### PR DESCRIPTION
**Pull request by hemildesai**
Refactors init of dtensor policy v2 as part of #1589. Depends on #1695 

### Issues
#1589 

Nightly links:
dpo - https://wandb.ai/nvidia/nemo-rl/runs/zx0io3io?nw=nwuserhemild
grpo moonlight - https://wandb.ai/nvidia/nemo-rl/runs/elsaxv44?nw=nwuserhemild
grpo qwen - https://wandb.ai/nvidia/nemo-rl/runs/kso90996?nw=nwuserhemild
sft gpt-oss - https://wandb.ai/nvidia/ruit_personal_debug/runs/g1lpu7e1

## Summary by CodeRabbit

* **New Features**
  * Introduced automated model configuration and distributed training setup system with support for multiple parallelism strategies and optional adapters.
  * Enhanced checkpoint management for streamlined save/load operations.

* **Refactor**
  * Simplified model initialization pipeline from monolithic to staged workflows.
  * Consolidated model, optimizer, and scheduler loading processes.

* **Tests**
  * Added extensive unit test coverage for setup utilities and configuration validation.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: nemo_rl/models/policy/workers/dtensor_policy_worker_v2.py [hunk 1]
```diff
@@ -202,430 +175,113 @@ def __init__(
...
+                "model_repo_id": config["model_name"],
+                "dequantize_base_checkpoint": config.get(
                     "dequantize_base_checkpoint", False
                 ),
-                "is_peft": self.lora_enabled,
+                "is_peft": config["dtensor_cfg"]
+                .get("lora_cfg", {})
+                .get("enabled", False),
             },
         )
```
[the remark this answers is not in the document]
Comment (hemildesai) [reply nemo_rl/models/policy/workers/dtensor_policy_worker_v2.py:225]: Fixed.

File: nemo_rl/models/policy/workers/dtensor_policy_worker_v2.py [hunk 2]
```diff
@@ -202,430 +175,112 @@ def __init__(
...
-
-        # Explicitly set NCCL_CUMEM_ENABLE to 1 to avoid the P2P initialization error for PyNCCLCommunicator.
-        # See https://github.com/NVIDIA-NeMo/RL/issues/564 for more details.
-        if not self.is_generation_colocated:
-            os.environ["NCCL_CUMEM_ENABLE"] = "1"
-
-        # Disable dynamo autotune_local_cache to avoid crash when there's already a cache
-        # with different order of node_bundles
-        configure_dynamo_cache()
+        is_vlm = processor is not None
```
Comment (yuki-97) [inline nemo_rl/models/policy/workers/dtensor_policy_worker_v2.py]: nit: how about just set `self.is_vlm` and `self.lora_enabled` here?
so that there's no need for `Additional derived attributes` below, also we can just use `"is_peft": self.lora_enabled`.

File: nemo_rl/models/policy/workers/dtensor_policy_worker_v2.py [hunk 3]
```diff
@@ -202,430 +175,112 @@ def __init__(
...
-            self.is_generation_colocated = config["generation"]["colocated"]["enabled"]
-
-        # Explicitly set NCCL_CUMEM_ENABLE to 1 to avoid the P2P initialization error for PyNCCLCommunicator.
-        # See https://github.com/NVIDIA-NeMo/RL/issues/564 for more details.
-        if not self.is_generation_colocated:
-            os.environ["NCCL_CUMEM_ENABLE"] = "1"
-
-        # Disable dynamo autotune_local_cache to avoid crash when there's already a cache
-        # with different order of node_bundles
-        configure_dynamo_cache()
```
Comment (yuki-97) [inline nemo_rl/models/policy/workers/dtensor_policy_worker_v2.py:222]: feels better to keep the two comments in `setup.py`, so that others can know why we need to do this.

File: nemo_rl/models/policy/workers/dtensor_policy_worker_v2.py [hunk 4]
```diff
@@ -202,430 +175,112 @@ def __init__(
...
-        cp_size_cfg = self.cfg["dtensor_cfg"]["context_parallel_size"]
-
-        # NeMoAutoModelForCausalLM uses flash_attention_2 by default
-        # so we need to set it to None if sequence packing is disabled
-        # https://github.com/NVIDIA-NeMo/Automodel/blob/7e748be260651349307862426c0c168cebdeeec3/nemo_automodel/components/_transformers/auto_model.py#L180
-        attn_impl = (
-            "flash_attention_2"
-            if (self.enable_seq_packing and cp_size_cfg == 1)
-            else ("sdpa" if cp_size_cfg > 1 else None)
-        )
```
Comment (yuki-97) [inline nemo_rl/models/policy/workers/dtensor_policy_worker_v2.py:268]: better to keep comments here as well.

File: nemo_rl/models/policy/workers/dtensor_policy_worker_v2.py (hunk 3 above)
Comment (terrykong) [reply nemo_rl/models/policy/workers/dtensor_policy_worker_v2.py:222]: agreed

**Comment by terrykong** [comment]
@hemildesai also, please run dtensor_v2 nightlies to check this refactor didn't introduce regression

File: nemo_rl/models/policy/workers/dtensor_policy_worker_v2.py (hunk 4 above)
Comment (hemildesai) [reply nemo_rl/models/policy/workers/dtensor_policy_worker_v2.py:268]: Done in [fix](https://github.com/NVIDIA-NeMo/RL/pull/1709/commits/895c1a4b44bb76b86b4db5725fe0bdba79a64ec8)

File: nemo_rl/models/policy/workers/dtensor_policy_worker_v2.py (hunk 3 above)
Comment (hemildesai) [reply nemo_rl/models/policy/workers/dtensor_policy_worker_v2.py:222]: Done in [fix](https://github.com/NVIDIA-NeMo/RL/pull/1709/commits/895c1a4b44bb76b86b4db5725fe0bdba79a64ec8)

File: nemo_rl/models/policy/workers/dtensor_policy_worker_v2.py (hunk 2 above)
Comment (hemildesai) [reply nemo_rl/models/policy/workers/dtensor_policy_worker_v2.py]: Done in [fix](https://github.com/NVIDIA-NeMo/RL/pull/1709/commits/895c1a4b44bb76b86b4db5725fe0bdba79a64ec8)
